src/Persistence/Collections/ControlPropertiesCollection.cs

```diff
@@ -79,6 +79,14 @@ public void Add(Tuple<string, string> keyValue)
         _properties.Add(keyValue.Item1, new ControlPropertyValue(keyValue.Item2));
     }
 
+    internal void Set(string key, ControlPropertyValue value)
```

Comment:
what if the key is not present? I'd suggest to remove this method and handle the logic via the `[]` operator instead

Reply:
It doesn't matter if the key is not present - it is either added or overwritten

Reply:
true, but now we have even another way of `Add`-ing a property. "Set" is a pretty generic name and could be interpreted as an upsert operation. At this point I'd suggest to update the class interface and rename all the `Add` to `Set` to avoid confusion

Reply:
> could be interpreted as an upsert operation

Correct, that's what this is.  The deserialization needs to add the ZIndex, and overwrite any existing ZIndex already there.